src/thunderbots/software/backend/backend_factory.cpp

```diff
@@ -0,0 +1,62 @@
+#include "backend/backend_factory.h"
+
+#include <exception>
+
+BackendRegistry& BackendFactory::getMutableRegistry()
+{
+    static BackendRegistry instance;
+    return instance;
+}
+
+const BackendRegistry& BackendFactory::getRegistry()
+{
+    return BackendFactory::getMutableRegistry();
+}
+
+std::vector<std::string> BackendFactory::getRegisteredBackendNames()
+{
+    std::vector<std::string> names;
+
+    for (auto it = BackendFactory::getRegistry().begin();
+         it != BackendFactory::getRegistry().end(); it++)
+    {
+        names.emplace_back(it->first);
+    }
+    return names;
+}
+
+std::vector<std::function<std::unique_ptr<Backend>()>>
+BackendFactory::getRegisteredBackendConstructors()
+{
+    std::vector<std::function<std::unique_ptr<Backend>()>> constructors;
+
+    for (auto it = BackendFactory::getRegistry().begin();
+         it != BackendFactory::getRegistry().end(); it++)
+    {
+        constructors.emplace_back(it->second);
+    }
+    return constructors;
+}
+
+void BackendFactory::registerBackend(
+    std::string backend_name, std::function<std::unique_ptr<Backend>()> backend_creator)
+{
+    BackendFactory::getMutableRegistry().insert(
```

Comment:
Definitely. Since this is the same as what we do for Play's, I'm inclined to file this as a general improvement that we should put in a "GenericFactory".

Reply:
Sounds good. I added to the General Factory notes so I remember to include this point when we make a ticket.